lib/read.js

```diff
@@ -65,6 +67,50 @@ class ReadElastic extends AdapterRead {
         return utils.default_config_property_for_id(this.id, property);
     }
 
+    _check_options_vs_mapping() {
+        if (this.checked_options_vs_mapping_already) { return Promise.resolve(); }
+        this.checked_options_vs_mapping_already = true;
+
+        return elastic.mapping_for_id(this.id, this.index, this.type)
+            .then((mapping) => {
+                _.each(mapping, (mapping_object, index) => {
+                    var index_mapping = mapping_object.mappings || {};
+                    _.each(index_mapping, (props_object, type) => {
+                        var properties = props_object.properties;
+                        this._warn_if_missing(properties, this.timeField, index, type);
+
+                        this.filtered_fields.forEach((field) => {
+                            this._warn_if_analyzed(properties, field, index, type);
+                            this._warn_if_missing(properties, field, index, type);
+                        });
+
+                        var reduce_by_fields = this.es_opts.aggregations &&
```

Comment:
nit, just a preference here so only change if you agree: 
I like `var y = a  || b` but  `var y = a && c  || b` isn't as clear to read as it is clever. 
Maybe use this instead:
```
var reduce_by_fields = [];
if (this.es_opts.aggregations) {
     reduce_by_fields = this.es_opts.aggregations.grouping;
}
```